pkg/logs/input/container/kubernetes_scanner.go

```diff
@@ -168,13 +168,43 @@ const kubernetesIntegration = "kubernetes"
 
 // getSource returns a new source for the container in pod
 func (s *KubeScanner) getSource(pod *kubelet.Pod, container kubelet.ContainerStatus) *config.LogSource {
-	return config.NewLogSource(s.getSourceName(pod, container), &config.LogsConfig{
-		Type:    config.FileType,
-		Path:    s.getPath(pod, container),
-		Source:  kubernetesIntegration,
-		Service: kubernetesIntegration,
-		Tags:    s.getTags(container),
-	})
+	var cfg *config.LogsConfig
+	if annotation := s.getAnnotation(pod, container); annotation != "" {
+		cfg, err := config.Parse(annotation)
+		if err != nil {
+			log.Warnf("Invalid pod annotation for pod %v, container %v: %v", pod.Metadata.Name, container.Name, err)
+			return nil
+		}
+	} else {
+		cfg = &config.LogsConfig{
+			Source:  kubernetesIntegration,
+			Service: kubernetesIntegration,
+		}
+	}
+	cfg.Type = config.FileType
+	cfg.Path = s.getPath(pod, container)
+	cfg.Tags = append(cfg.Tags, s.getTags(container)...)
+	return config.NewLogSource(s.getSourceName(pod, container), cfg)
+}
+
+// configPath refers to the configuration that can be passed over a pod annotation,
+// this feature is commonly named 'ad' or 'autodicovery'.
+// The pod annotation must respect the format: ad.datadoghq.com/<container_name>.logs: [{...}]
+const (
+	configPathPrefix = "ad.datadoghq.com"
```

Comment:
It feels weird that this is located in `kubernetes`.

Reply:
agreed but this should be removed soon when we'll be fully plugged to the AD.